### PR TITLE
[DF] Early-quit event loop for RDataSource+Range 

### LINKED ISSUE #6455: [DF] RDataSource does not early-quit event loops when all Ranges are exhausted
## Bug description

When running over TTrees, RDF event loops that exhaust all entry ranges specified via `df.Range(..)` quit the event loop early as a performance optimization. RDataSource-based event loops do not, and instead run until the end of the dataset in all cases (performing no processing after Ranges have been exhausted). This yields correct results but wastes CPU cycles.

### PR DESCRIPTION
**Pull request by eguiraud**
RDataSource event loops did not check whether Ranges had been exhausted
and always looped until the end of the dataset in all cases.
Event processing was a no-op after Ranges had been exhausted so results
were correct but runtimes were longer than necessary.

This fixes #6455.

### CODE REVIEW

File: tree/dataframe/inc/ROOT/RTrivialDS.hxx [hunk 1]
```diff
@@ -34,6 +34,8 @@ protected:
 
 public:
    RTrivialDS(ULong64_t size, bool skipEvenEntries = false);
+   // This ctor produces a data-source that returns infinite entries
```
Comment (hageboeck) [inline tree/dataframe/inc/ROOT/RTrivialDS.hxx:37]:
```suggestion
   /// This ctor produces a data-source that returns infinite entries
```

Comment (hageboeck) [reply tree/dataframe/inc/ROOT/RTrivialDS.hxx:37]: Is doxygen documentation desired for this class?

**Review by hageboeck — APPROVED** [review]
I cannot follow, but the code at least looks reasonable.

File: tree/dataframe/inc/ROOT/RTrivialDS.hxx (hunk 1 above)
Comment (eguiraud) [reply tree/dataframe/inc/ROOT/RTrivialDS.hxx:37]: Not really, RTrivialDS is just there for testing purposes. Honestly I don't even know why it's in the public namespace. Buuuut an extra slash doesn't hurt!

**Pull request merged**